lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -2861,6 +2861,62 @@ bool FoldChannelShuffle::run(Function *F, const CompilationContext &cctx) {
   return changed;
 }
 
+// Fold Tile -> Add into BatchedAdd wherever applicable.
+bool FoldTileAddIntoBatchedAdd::run(Function *F,
+                                    const CompilationContext &cctx) {
+  LOG_SCOPE(F->getLogContext(), getName());
+
+  bool changed = false;
+  auto &nodes = F->getNodes();
+  for (auto &node : nodes) {
+    auto *addNode = dyn_cast<AddNode>(&node);
+    if (!addNode) {
+      continue;
+    }
+
+    NodeValue batchNode, addedNode;
+    auto &LHS = addNode->getLHS();
+    auto &RHS = addNode->getRHS();
+    auto *tileNode = dyn_cast<TileNode>(LHS);
+
+    // Check if LHS is a tile.
+    if (tileNode) {
+      batchNode = RHS;
+      addedNode = tileNode->getInput();
+    }
+    // Check if RHS is a tile.
+    else if ((tileNode = dyn_cast<TileNode>(RHS))) {
```

Comment:
```suggestion
    else if (tileNode = dyn_cast<TileNode>(RHS)) {
```
I think there are extra braces here? I also remember getting some compiler error/warning when I tried something like this without the extra braces so I might be wrong here.

Reply:
Extra braces are to make the compiler not show warnings like if it was meant to be ==